python/python/glide/async_commands/core.py

```diff
@@ -1232,3 +1234,104 @@ async def zrem(
             int,
             await self._execute_command(RequestType.Zrem, [key] + members),
         )
+
+    @dataclass
+    class PubSubMsg:
+        """ Describes incomming message """
+        message: str
+        """
+        Incomming message
+        """
+        channel: str
+        """
+        Name of an channel that triggered the message
+        """
+        pattern: Optional[str]
+        """
+        Pattern that triggered the message
+        """
+
+    class ChannelModes(Enum):
+        """ Describes subsciption modes. """
+        Exact = 1
+        """ Use exact channel names """
+        Pattern = 2
+        """ Use channel name patterns """
+        Sharded = 3
+        """ Use sharded PUBSUB. See  https://redis.io/docs/interact/pubsub/#sharded-pubsub for more details """
+
+    async def subscribe(self,
+                        channels_or_patterns: Set[str],
+                        channel_mode: ChannelModes,
+                        callback: Callable[[PubSubMsg, Optional[Any]]],
+                        context: Optional[Any]) -> None:
```

Comment:
why returning None and not OK?

Reply:
dont we use Exceptions to signal errors?

Reply:
just saw redis' doc "When successful, this command doesn't return anything", so ignore this comment

Reply:
done